### PR TITLE
Add health check task

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

File: content/docs/tasks/security/app-health-check.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+There are three options for liveness and readiness probe in Kubernetes: 1) command; 2) http request; 3) tcp request. In this task, we provide examples for the first two options with Istio mutual TLS enabeld and disabled, respecitively. We will cover the third option soon.
+
+## Before you begin
+
+* Understand [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/), Istio [authentication policy](/docs/concepts/security/authn-policy/) and [mutual TLS authentication](/docs/concepts/security/mutual-tls/) concepts.
+
+* Have a Kubernetes cluster with Istio installed, without global mutual TLS enabled (e.g use `install/kubernetes/istio-demo.yaml` as described in [installation steps](/docs/setup/kubernetes/quick-start/#installation-steps), or set `global.mtls.enabled` to false using [Helm](/docs/setup/kubernetes/helm-install/)).
+
+## Liveness and Readiness Probe with Command Option
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: See https://istio.io/about/contribute/style-guide/#only-capitalize-the-first-letter-of-headings for this and other headings.

File: content/docs/tasks/security/app-health-check.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+This task shows how to use [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/) for health check on Istio services.
+
+There are three options for liveness and readiness probe in Kubernetes: 1) command; 2) http request; 3) tcp request. In this task, we provide examples for the first two options with Istio mutual TLS enabeld and disabled, respecitively. We will cover the third option soon.
+
+## Before you begin
+
+* Understand [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/), Istio [authentication policy](/docs/concepts/security/authn-policy/) and [mutual TLS authentication](/docs/concepts/security/mutual-tls/) concepts.
+
+* Have a Kubernetes cluster with Istio installed, without global mutual TLS enabled (e.g use `install/kubernetes/istio-demo.yaml` as described in [installation steps](/docs/setup/kubernetes/quick-start/#installation-steps), or set `global.mtls.enabled` to false using [Helm](/docs/setup/kubernetes/helm-install/)).
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: Make a link to the YAML. The link should use the special marker {{< branch_name >}} to refer to the current branch in the GitHub link.

File: content/docs/tasks/security/app-health-check.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+## Liveness and Readiness Probe with Command Option
+
+In this section, we first show configure health check with mutual TLS disabled and then show the health check with mutual TLS enabled since currently your cluster has mutual TLS disabled.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: show how to configure health checking when mutual TLS is disabled, then how it works when mutual TLS is enabled.

File: content/docs/tasks/security/app-health-check.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+### Mutual TLS Disabled
+
+Run this command to deploy [liveness](https://github.com/istio/istio/blob/{{<branch_name>}}/samples/health-check/liveness-command.yaml) on default namespace:
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: on default -> in the default

File: content/docs/tasks/security/app-health-check.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+Wait for a minute and check the pod status
+```command
+$ kubectl get pod
+NAME                             READY     STATUS    RESTARTS   AGE
+liveness-6857c8775f-zdv9r        2/2       Running   0           1m
+```
+
+The number '0' in 'RESTARTS' column means liveness probe works fine. Readiness probe works in the same way and you can modify liveness-command.yaml accordingly to try it yourself.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: in 'RESTARTS' -> in the 'RESTARTS'

Comment (geeknoid) [reply content/docs/tasks/security/app-health-check.md]: probe -> probes

File: content/docs/tasks/security/app-health-check.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+### Mutual TLS Enabled
+
+Run this command to enable mutual TLS for services on default namespace.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: on -> in the

File: content/docs/tasks/security/app-health-check.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+Run this command to re-deploy the service:
+
+```command
+$ kubectl delete -f <(istioctl kube-inject -f @samples/health-check/liveness-command.yaml@)
+$ kubectl apply -f <(istioctl kube-inject -f @samples/health-check/liveness-command.yaml@)
+```
+
+And repeat the same steps as in above subsection to verify that liveness probe works.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: as in the above section
liveness -> the liveness

File: content/docs/tasks/security/app-health-check.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+## Liveness and Readiness Probe with Http Request Option
+
+This section shows how to configure health check with http request option.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: check -> checking
with http -> with the HTTP

File: content/docs/tasks/security/app-health-check.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+metadata:
+  name: "example-1"
+  namespace: "default"
+spec:
+  peers:
+  - mtls:
+EOF
+```
+
+Run this command to deploy [liveness-http](https://github.com/istio/istio/blob/{{<branch_name>}}/samples/health-check/liveness-http.yaml) on default namespace:
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: on -> in the

File: content/docs/tasks/security/app-health-check.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+```command
+$ kubectl apply -f <(istioctl kube-inject -f @samples/health-check/liveness-http.yaml@)
+```
+
+Wait for a minute and check the pod status to make sure liveness probe works with RESTARTS column to be '0'.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: liveness -> the liveness

File: content/docs/tasks/security/app-health-check.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+```command
+$ kubectl get pod
+NAME                             READY     STATUS    RESTARTS   AGE
+liveness-http-975595bb6-5b2z7c   2/2       Running   0           1m
+```
+
+### Mutual TLS Enabled
+
+Run this command to enable mutual TLS for services on default namespace.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: on -> in the

File: content/docs/tasks/security/app-health-check.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+kind: "Policy"
+metadata:
+  name: "example-1"
+  namespace: "default"
+spec:
+  peers:
+EOF
+```
+
+Run this command to re-deploy the service:
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: this command -> these commands

File: content/docs/tasks/security/app-health-check.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+```command
+$ kubectl delete -f <(istioctl kube-inject -f @samples/health-check/liveness-http.yaml@)
+$ kubectl apply -f <(istioctl kube-inject -f @samples/health-check/liveness-http.yaml@)
+```
+
+Wait for a minute and check the pod status to make sure liveness probe works with RESTARTS column to be '0'.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: liveness -> the liveness

File: content/docs/tasks/security/app-health-check.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+Note that the image in [liveness-http](https://github.com/istio/istio/blob/{{<branch_name>}}/samples/health-check/liveness-http.yaml) exposes two ports: 8001 and 8002 ([source code](https://github.com/istio/istio/blob/{{<branch_name>}}/samples/health-check/server.go)). In this deployment, port 8001 serves the regular traffic while port 8002 is used for liveness probe. Since Istio proxy only intercepts ports that are explicitly declared at 'containerPort' fields, traffic to 8002 port will bypass Istio proxy no matter Istio mutual TLS is enabled or not. However, if we use port 8001 for both regualr traffic and liveness probe, health check will fail when mutual TLS is enabled since the http request is sent from Kubelet, which does not send client certificate to the liveness-http service.
+
+Note: we will have a permissive mode in AuthenticationPolicy soon so a port can take both http and mutual TLS traffic. With this mode, liveness probe should work even when regular traffic and health check are on the same port. This could be a workaround for liveness check even though mutual TLS is not actually enforced. In the meantime, we are actively working on a complete solution.
```
Comment (geeknoid) [inline content/docs/tasks/security/app-health-check.md]: Let's remove this paragraph per the style guide's recommendation about future promises.

Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done

File: content/docs/tasks/security/app-health-check.md (hunk 13 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md (hunk 12 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md (hunk 11 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md (hunk 10 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+keywords: [security,health-check]
+---
+
+This task shows how to use [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/) for health check on Istio services.
+
+There are three options for liveness and readiness probe in Kubernetes: 1) command; 2) http request; 3) tcp request. In this task, we provide examples for the first two options with Istio mutual TLS enabeld and disabled, respecitively. We will cover the third option soon.
+
+## Before you begin
+
+* Understand [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/), Istio [authentication policy](/docs/concepts/security/authn-policy/) and [mutual TLS authentication](/docs/concepts/security/mutual-tls/) concepts.
```
[the remark this answers is not in the document]
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md [hunk 16]
```diff
@@ -0,0 +1,132 @@
+---
+title: Health Check on Istio Services
+description: Shows how to do health check for Istio services.
+weight: 65
+keywords: [security,health-check]
+---
+
+This task shows how to use [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/) for health check on Istio services.
+
+There are three options for liveness and readiness probe in Kubernetes: 1) command; 2) http request; 3) tcp request. In this task, we provide examples for the first two options with Istio mutual TLS enabeld and disabled, respecitively. We will cover the third option soon.
```
[the remark this answers is not in the document]
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md [hunk 17]
```diff
@@ -0,0 +1,132 @@
+---
+title: Health Check on Istio Services
+description: Shows how to do health check for Istio services.
+weight: 65
+keywords: [security,health-check]
+---
+
+This task shows how to use [Kubernetes liveness and readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/) for health check on Istio services.
```
[the remark this answers is not in the document]
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md [hunk 18]
```diff
@@ -0,0 +1,132 @@
+---
+title: Health Check on Istio Services
+description: Shows how to do health check for Istio services.
```
[the remark this answers is not in the document]
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md [hunk 19]
```diff
@@ -0,0 +1,132 @@
+---
+title: Health Check on Istio Services
```
[the remark this answers is not in the document]
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md (hunk 1 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

File: content/docs/tasks/security/app-health-check.md (hunk 2 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md (hunk 3 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: Done.

File: content/docs/tasks/security/app-health-check.md (hunk 4 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

File: content/docs/tasks/security/app-health-check.md (hunk 5 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

File: content/docs/tasks/security/app-health-check.md (hunk 6 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

File: content/docs/tasks/security/app-health-check.md (hunk 7 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

File: content/docs/tasks/security/app-health-check.md (hunk 8 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

File: content/docs/tasks/security/app-health-check.md (hunk 9 above)
Comment (wattli) [reply content/docs/tasks/security/app-health-check.md]: done.

**Comment by wattli** [comment]
Thanks Martin for the detailed comments!

**Comment by wattli** [comment]
And this task is moved to traffic management section.

**Comment by wattli** [comment]
@geeknoid , do you know how to fix the spelling check error in circleci for 'containerPort'? https://circleci.com/gh/istio/istio.github.io/2664?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by geeknoid** [comment]
See here on spelling: https://github.com/istio/istio.github.io#linting

**Comment by wattli** [comment]
@geeknoid , Thanks, done,

**Comment by wattli** [comment]
@geeknoid , please take a look, thanks!

**Pull request merged**